tests/test_unit.py

```diff
@@ -2205,6 +2205,29 @@ def testParseResultsPickle(self):
                 ),
             )
 
+    def testMatchOnlyAtCol(self):
+        """successfully use matchOnlyAtCol helper function"""
+        from pyparsing import nums, ZeroOrMore, Word, matchOnlyAtCol
+
+        expr = Word(nums)
+        expr.setParseAction(matchOnlyAtCol(5))
+        largerExpr = ZeroOrMore(Word("A")) + expr + ZeroOrMore(Word("A"))
+
+        res = largerExpr.parseString("A A 3 A")
+        print(res.dump())
+
+    def testMatchOnlyAtColErr(self):
+        """raise a ParseException in matchOnlyAtCol with incorrect col"""
+        from pyparsing import nums, ZeroOrMore, Word, matchOnlyAtCol
+
+        expr = Word(nums)
+        expr.setParseAction(matchOnlyAtCol(1))
+        largerExpr = ZeroOrMore(Word("A")) + expr + ZeroOrMore(Word("A"))
+
+        with self.assertRaises(ParseException) as ar:
+            res = largerExpr.parseString("A A 3 A")
+        print(type(ar.exception).__name__, str(ar.exception))
```

Comment:
Add this method to Test2_WithoutPackrat class, and all assertRaises will print out the exception info (will also need to import contextlib at the top):

    @contextlib.contextmanager
    def assertRaises(self, expected_exception_type, msg=None):
        """
        Simple wrapper to print out the exceptions raised after assertRaises
        """
        try:
            with super().assertRaises(expected_exception_type, msg=msg) as ar:
                yield
        finally:
            if getattr(ar, 'exception', None) is not None:
                print('Raised expected exception: {}: {}'.format(type(ar.exception).__name__,
                                                                 str(ar.exception)))
            else:
                print('Expected {} exception not raised'.format(expected_exception_type.__name__))

Reply:
I like this. One note, it no longer returns the exception so using the following syntax creates a None object for ar:

    with self.assertRaises(ParseException) as ar:

That may not be necessary anymore now that the print is built in but I'm mentioning it in case it wasn't an intended effect.

Reply:
Omission on my part, can you correct by adding "return ar" after the "finally" clause?

Reply:
Oh, and you should be able to do `self.assertRaisesParseException()` instead of `self.assertRaises(ParseException)`. Right now the behavior is equivalent, but I might be able to add some ParseException-specific stuff in that method.

Reply:
Yes, adding that works.